### PR TITLE
Amp 847 face

### PR DESCRIPTION
**Pull request by yingfeng-iu**
Ready for review

### CODE REVIEW

File: src/main/java/edu/indiana/dlib/amppd/repository/ItemSupplementRepository.java [hunk 1]
```diff
@@ -8,4 +10,7 @@
 @CrossOrigin(origins = "*")
 @RepositoryRestResource(collectionResourceRel = "itemSupplements", path = "itemSupplements")
 public interface ItemSupplementRepository extends SupplementRepository<ItemSupplement> {
+	
+	List<ItemSupplement> findByItemIdAndName(Long itemId, String name);
```
Comment (dfische3) [inline src/main/java/edu/indiana/dlib/amppd/repository/ItemSupplementRepository.java:14]: @yingfeng-iu If we have the ID, why do we need the name as well?

File: src/main/java/edu/indiana/dlib/amppd/service/impl/MediaServiceImpl.java [hunk 2]
```diff
@@ -104,6 +107,41 @@ public MediaServiceImpl(AmppdUiPropertyConfig amppduiConfig) {
...
+		}		
+		
+		// there should be exactly one supplement found, as supplement is unique by name within its parent's scope
+		if (supplements != null && supplements.size() == 1) {
+			supplement = supplements.get(0);
+		}	
+		
+		// if the exact supplement is found, resolve its pathname to the absolute path
+		if (supplement != null) {
+			pathname = fileStorageService.resolve(supplement.getPathname()).toString();
```
Comment (dfische3) [inline src/main/java/edu/indiana/dlib/amppd/service/impl/MediaServiceImpl.java:139]: Can this move to between line 134-135 and remove the null check?

**Review by dfische3 — APPROVED** [review]
Looks good.  Just had a few minor questoins

File: src/main/java/edu/indiana/dlib/amppd/repository/ItemSupplementRepository.java (hunk 1 above)
Comment (yingfeng-iu) [reply src/main/java/edu/indiana/dlib/amppd/repository/ItemSupplementRepository.java:14]: The method is using itemId, not ID of the supplement. By the way, I believe itemID together with supplement name should uniquely identify the item supplement (just like item ID plus primaryfile name uniquely identify a primaryfile). If this is not in the batch ingest validation, please let me know.

File: src/main/java/edu/indiana/dlib/amppd/service/impl/MediaServiceImpl.java (hunk 2 above)
Comment (yingfeng-iu) [reply src/main/java/edu/indiana/dlib/amppd/service/impl/MediaServiceImpl.java:139]: Good point. will do! I'm also adding some unit tests.

**Pull request merged**